flight/Modules/Actuator/actuator.c

```diff
@@ -191,76 +214,34 @@ static void actuatorTask(void* parameters)
 	{
 		PIOS_WDG_UpdateFlag(PIOS_WDG_ACTUATOR);
 
-		// Wait until the ActuatorDesired object is updated
-		bool rc = PIOS_Queue_Receive(queue, &ev, FAILSAFE_TIMEOUT_MS);
+		bool rc=false;
 
-		/* Process settings updated events even in timeout case so we always act on the latest settings */
-		if (actuator_settings_updated) {
-			actuator_settings_updated = false;
-			ActuatorSettingsGet (&actuatorSettings);
-			actuator_update_rate_if_changed (&actuatorSettings, false);
+		// If it's our first iteration, fall down to getting the
+		// config and enabling failsafe.
+		if (!first) {
```

Comment:
That would cause undefined behavior unless actuatorSettings is global, because it means that we could be running the loop when we get that update callback and internal data structures change.  Unless something is declared "volatile", the optimizer is allowed to make optimizations on the assumption that actuatorSettings doesn't change within code unless it is passed to something that could change it within that thread of execution (that is, a function that changes it or a function in another module that escapes analysis).   Undefined behavior means the code can do literally anything, not necessarily something one would suspect from inspecting the input source (e.g. run "both halves" of an if/else statement, or..).

Declaring it volatile is really bad for performance (generates frequent loads) and even so takes a lot of analysis on conditionals that state changes don't do something bad.